File: BCC221-POO/TPJAVA/TPJava/src/tpjava/Veiculo.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package tpjava;
import java.util.ArrayList;

/**
 *
 * @author nicolas
 */
public class Veiculo {
    private String placa;
    private double quilometragem;
    private ArrayList<String> historicoAjuste;

    public Veiculo(String placa, double quilometragem) {
        this.placa = placa;
        this.quilometragem = quilometragem;
    }

    public String getPlaca() {
        return placa;
    }

    public void setPlaca(String placa) {
        this.placa = placa;
    }

    public double getQuilometragem() {
        return quilometragem;
    }

    public void setQuilometragem(double quilometragem) {
        this.quilometragem = quilometragem;
    }

    public void addHistorico(String historico){
        this.historicoAjuste.add(historico);
    }
    
    public void getHistorico(int index){
        this.historicoAjuste.get(index);
    }
    
    public void showHistorico(){
        historicoAjuste.forEach((_item) -> {
            System.out.println("Resposta do Mecânico: " + _item);
        });
    }
    
}

File: BCC221-POO/TPC++/classes_cpp/veiculo.cpp
#include "../headers/veiculo.hpp"

using namespace std;

Veiculo::~Veiculo() {}

Veiculo::Veiculo(string placa, float quilometragem){
    this->placa = placa;
    this->quilometragem = quilometragem;
}

void Veiculo::addHistAjust(string ajuste){
    this->hist_ajust.push_back(ajuste);
}

string Veiculo::getPlaca(){
    return this->placa;
}

float Veiculo::getQuilometragem(){
    return this->quilometragem;
}

string Veiculo::getHist(){
    string todas_respostas = "";

    for (int i = 0; i < this->hist_ajust.size(); i++)
    {
        todas_respostas = todas_respostas + this->hist_ajust[i] + "\n";
    }
    
    return todas_respostas;
}

int Veiculo::getTamHist(){
    return this->hist_ajust.size();
}